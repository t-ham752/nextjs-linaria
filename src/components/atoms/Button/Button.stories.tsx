import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import { Button } from "./";

import { screen, userEvent } from "@storybook/testing-library";

export default { component: Button } as ComponentMeta<typeof Button>;

export const Default: ComponentStoryObj<typeof Button> = {
  args: { children: "exe", ariaLabel: "Open modal" },
};

Default.play = async () => {
  const button = screen.getByRole("button");

  await userEvent.click(button);

  // このテストはいつ走らせるのか
};
