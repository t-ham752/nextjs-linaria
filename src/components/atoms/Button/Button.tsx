import { styled } from "linaria/react";
import React from "react";
// ______________________________________________________
//
export type Props = {
  /**
   * ボタンに表示する文字
   */
  children: string;
  /**
   * フォントサイズ
   */
  fontSize: string;
  /**
   * aria-label
   */
  ariaLabel?: string;
};
// ______________________________________________________
//
export const Button: React.FC<Props> = ({
  children = "action",
  fontSize = "32px",
  ariaLabel,
}: Props) => {
  return (
    <StyledButton fontSize={fontSize} aria-label={ariaLabel}>
      {children}
    </StyledButton>
  );
};

const StyledButton = styled.button<Props>`
  font-size: ${(props) => props.fontSize ?? "14px"};
`;
