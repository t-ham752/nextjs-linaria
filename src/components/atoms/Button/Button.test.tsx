/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";

import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

import * as stories from "./Button.stories";
// ______________________________________________________
//
const { Default } = composeStories(stories);

describe("atoms/Button", () => {
  test("Default", () => {
    render(<Default />);
    expect(screen.getByRole("button")).toHaveTextContent("exe");
    expect(screen.getByRole("button")).toHaveAttribute("font-size", "32px");
    expect(
      screen.getByRole("button", { name: "Open modal" })
    ).toBeInTheDocument();
    expect(screen.getByLabelText("Open modal")).toBeInTheDocument();
    expect(screen.getByRole("button")).toMatchSnapshot();
  });
});
