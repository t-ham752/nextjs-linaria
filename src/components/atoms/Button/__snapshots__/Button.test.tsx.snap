// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Button Default 1`] = `
<button
  aria-label="Open modal"
  class="s8sdw4g"
  font-size="32px"
  style="--s8sdw4g-0: 32px;"
>
  exe
</button>
`;
